src/NHibernate/Mapping/Table.cs

```diff
@@ -972,30 +972,34 @@ public override string ToString()
 			return buf.ToString();
 		}
 
-		public void ValidateColumns(Dialect.Dialect dialect, IMapping mapping, ITableMetadata tableInfo)
+		public List<HibernateException> ValidateColumns(Dialect.Dialect dialect, IMapping mapping, ITableMetadata tableInfo)
 		{
 			IEnumerable<Column> iter = ColumnIterator;
+			List<HibernateException> exceptions = new List<HibernateException>();
 			foreach (Column column in iter)
 			{
 				IColumnMetadata columnInfo = tableInfo.GetColumnMetadata(column.Name);
 
 				if (columnInfo == null)
 				{
-					throw new HibernateException(string.Format("Missing column: {0} in {1}", column.Name,
-															   dialect.Qualify(tableInfo.Catalog, tableInfo.Schema, tableInfo.Name)));
+					exceptions.Add(new HibernateException(string.Format("Missing column: {0} in {1}", column.Name,
+															   dialect.Qualify(tableInfo.Catalog, tableInfo.Schema, tableInfo.Name))));
```

Comment:
Here lies a bug due to the change. A throw stops code execution, protecting in this case the code from reaching a null ref condition. You should make sure that everywhere you have replaced a throw by just an add in a list, the code following that will support the state for which an exception was supposed to be thrown or will be appropriately bypassed.